Slight fix to UVLModelLoader

shartte opened the pull request:
This fixes an issue in UVLModelLoader where it would not close the resources it opens to inspect them for the uvlMarker tag.

In addition, it optimizes loading times slightly by skipping Vanilla resources.

shartte committed:
Fixes resources not being closed by UVLModelLoader and optimizes loading times by skipping Vanilla resources in the loader.

Elix-x commented:
If you're going for it, then move IO stream into the try and close it once done .

Also, there's no point in skipping vanilla. We won't gain much and RP makers won't be able to make glowing vanilla blocks.

shartte commented:
> If you're going for it, then move IO stream into the try and close it once done .
> The resource closes the IO stream. There is a reason it implements Closeable.

I did a quick check. The no-vanilla improvement saves us from opening up and parsing 4006 files on startup.

shartte commented:
But there's another issue with the current implementation. It's not working as a drop-in replacement. It just does nothing. I'll check whats going on.

Elix-x commented:
> I did a quick check. The no-vanilla improvement saves us from opening up and parsing 4006 files on startup.

<nbsp>

> RP makers won't be able to make glowing vanilla blocks.

<nbsp>

> But there's another issue with the current implementation. It's not working as a drop-in replacement. It just does nothing. I'll check whats going on.

What do you mean? It does work for me.

shartte commented:
I don't know yet, it's working (everything shows up). But the UVLModelLoader skips the base model because it cannot open the file (FileNotFoundException).
I am investigating.

Elix-x commented:
Then it has nothing to do with uvl loader (check paths).

shartte commented:
@elix-x Yeah no, it has nothing to do with the paths since they work with the vanilla loader and the models show up correctly ingame. They're just not using the UVL extensions.
Can you please check if it's really working for you with your test mod? Even if I start your Test Mod, the block works, but it's not lit up at night.

_edit_ ![Example](https://i.imgur.com/2eSVrZo.jpg)

Elix-x commented:
Now go underground, place it and close off all light sources.

And i'm looking into why glow has decreased.

shartte commented:
If accepts() never returns true, it simply wont be active.

Fixed it by appending the missing .json to the path it tests:

```
        ResourceLocation actualLoc = new ResourceLocation( modelLocation.getResourceDomain(), modelLocation.getResourcePath() + ".json" );

        try( IResource resource = Minecraft.getMinecraft().getResourceManager().getResource( actualLoc ) )
        {
            return gson.fromJson( new InputStreamReader( resource.getInputStream() ), UVLMarker.class ).uvlMarker;
        }
        catch( IOException e )
        {
            return false;
        }
```

Now it does apply the UVLModelLoader, but yeah... I think your vanilla RP idea is not going to work:
_edit_ and neither is extracting the controller base model.

```
Caused by: java.lang.IllegalStateException: vanilla model 'net.minecraft.client.renderer.block.model.ModelBlock@65d7108d' can't have non-vanilla parent
    at net.minecraftforge.client.model.ModelLoader$VanillaModelWrapper.getTextures(ModelLoader.java:485) ~[ModelLoader$VanillaModelWrapper.class:?]
```

shartte committed:
Append .json to the path that is tested so it actually finds the model.
Moved the return into the exception block so the exception can be inspected for the breakpoint.

shartte commented:
I amended the PR to include the .json suffix.

Elix-x commented:
Alright, i'll look into how that can be avoided.

_Personal note: never ever test lighting in bright room with sun reflecting on to your monitor :D..._

shartte commented:
@elix-x What's the status on this one? I would like to use UVL for the drives and can't because this isn't fixed yet.

Elix-x commented:
_Waiting for something i don't have full control of..._

shartte commented:
@elix-x And what would that be?

Elix-x commented:
_Something i have partial control of :D_

shartte commented:
This is no longer relevant
